ui/app/src/components/Notifications.vue

```diff
@@ -1,18 +1,102 @@
 <script lang="ts">
 import { computed, defineComponent } from "vue";
-import { ref, Ref } from "@vue/reactivity"; /* eslint-disable-line */
+import { reactive, ref, Ref } from "@vue/reactivity"; /* eslint-disable-line */
 import { useCore } from "@/hooks/useCore";
+import { AppEvent } from "ui-core/src/api/NotificationService";
+
+// Message?
+type Notification = {
+  id?: string; // id would be used to remove timeout, may only need to be local type
+  type: "error" | "success" | "info";
+  message: string;
+  detail?: {
+    type: "etherscan" | "websocket" | "info";
+    message: string;
+  };
+  loader?: boolean;
+};
+
+// Visual Notifications are a view level system here we work out which ones are displayed to the user
+function parseEventToNotifications(event: AppEvent): Notification | null {
```

Comment:
This is cool. Should also give us more control how we want to display dispatch. For example, use the default popup in the corner or some other UI notification (like red text under an input).

Reply:
Exactly!!!